runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -117,48 +117,56 @@ func (q *MetricsViewComparison) Resolve(ctx context.Context, rt *runtime.Runtime
 }
 
 func (q *MetricsViewComparison) calculateMeasuresMeta() error {
-	measureMap := make(map[string]bool, len(q.Measures))
-	for _, m := range q.Measures {
-		measureMap[m.Name] = true
-	}
-
 	compare := !isTimeRangeNil(q.ComparisonTimeRange)
 
 	if !compare && len(q.ComparisonMeasures) > 0 {
 		return fmt.Errorf("comparison measures are provided but comparison time range is not")
 	}
 
-	comparisonMap := make(map[string]bool, len(q.ComparisonMeasures))
-
-	// backwards compatibility
 	if len(q.ComparisonMeasures) == 0 && compare {
-		comparisonMap = measureMap
-	}
-
-	for _, m := range q.ComparisonMeasures {
-		if !measureMap[m] {
-			return fmt.Errorf("comparison measure '%s' is not present in the metrics view '%s'", m, q.MetricsViewName)
+		// backwards compatibility
+		q.ComparisonMeasures = make([]string, len(q.Measures))
+		for i, m := range q.Measures {
+			q.ComparisonMeasures[i] = m.Name
+		}
+	} else {
+		for _, cm := range q.ComparisonMeasures {
+			found := false
+			for _, m := range q.Measures {
+				if m.Name == cm {
+					found = true
+					break
+				}
+			}
+			if !found {
+				return fmt.Errorf("comparison measure '%s' is not present in the measures list", cm)
+			}
 		}
-		comparisonMap[m] = true
 	}
 
-	err := validateSort(q.Sort, comparisonMap, compare)
+	err := validateSort(q.Sort, q.ComparisonMeasures, compare)
 	if err != nil {
 		return err
 	}
 
-	err = validateAlias(q.Aliases, comparisonMap)
+	err = validateMeasureAliases(q.Aliases, q.ComparisonMeasures, compare)
```

Comment:
Sorry, maybe it wasn't clear before, but why not use `q.measuresMeta` to do these validations? I.e. build it first, validate after. That way, it's just one map allocation and only one nested loop (for building `q.measuresMeta`).